hpx/runtime/threads/detail/scheduling_loop.hpp

```diff
@@ -420,6 +422,23 @@ namespace hpx { namespace threads { namespace detail
                             num_thread);
                         scheduler.SchedulingPolicy::do_some_work(num_thread);
                     }
+                    else if (HPX_UNLIKELY(state_val == pending_boost))
+                    {
+                        if (HPX_LIKELY(next_thrd == nullptr)) {
+                            // schedule other work
+                            scheduler.SchedulingPolicy::wait_or_add_new(
+                                num_thread, this_state.load() < state_stopping,
+                                idle_loop_count);
+                        }
+
+                        thrd->set_state(pending);
+
+                        // schedule this thread again immediately with boosted
+                        // priority
+                        scheduler.SchedulingPolicy::schedule_thread(thrd,
+                            num_thread, thread_priority_boost);
```

Comment:
Also, I don't think the argument that executing it right away reduces overheads measurably is a valid one. I anticipate that the `pending_boost` thread state will be used in rare cases, so it will not impact performance as such.